### DISCUSSION TITLE
Disable type-checking of emits?

### DISCUSSION THREAD

**Discussion by RobWalker** [johnsoncodehk/vue-tsc#69] [Q&A]
I have a component:
```
<ServerReport @row-double-clicked="onSelect">
```

Where `ServerReport` does **not** have a `defineEmits` declaration, and this gives an error.  Which is fair enough.

The problem is that `ServerReport` proxies all listeners down to a child component via a `v-bind="$attrs"`.  If I add a `defineEmits` to `ServerReport` directly then it breaks the event flow through this proxy.

See this [SFC Playground](https://sfc.vuejs.org/#eyJBcHAudnVlIjoiPHNjcmlwdCBzZXR1cD5cbiAgaW1wb3J0IHsgcmVmIH0gZnJvbSAndnVlJ1xuXG4gIGltcG9ydCBOZXN0ZWQgZnJvbSAnLi9OZXN0ZWQudnVlJztcbiAgXG4gIGNvbnN0IG9uRG9Tb21ldGhpbmcgPSAodikgPT4ge1xuICAgIGNvbnNvbGUubG9nKFwib25Eb1NvbWV0aGluZyBcIiArIHYpO1xuICB9XG48L3NjcmlwdD5cblxuPHRlbXBsYXRlPlxuICA8TmVzdGVkIEB1cGRhdGU6cmVjb3JkPVwib25Eb1NvbWV0aGluZ1wiPjwvTmVzdGVkPlxuPC90ZW1wbGF0ZT5cbiIsIklubmVyLnZ1ZSI6Ijx0ZW1wbGF0ZT5cbiAgPGJ1dHRvbiBAY2xpY2s9XCJvbkNsaWNrXCI+Y2xpY2sgbWU8L2J1dHRvbj5cbjwvdGVtcGxhdGU+XG5cbjxzY3JpcHQgc2V0dXA+XG4gIGNvbnN0IGVtaXQgPSBkZWZpbmVFbWl0cyhbXCJ1cGRhdGU6cmVjb3JkXCJdKTtcbiAgXG4gIGxldCBjb3VudCA9IDA7XG4gIFxuICBjb25zdCBvbkNsaWNrPSgpID0+IHtcbiAgICBjb25zb2xlLmxvZyhcIklubmVyLm9uQ2xpY2tcIik7XG4gICAgZW1pdChcInVwZGF0ZTpyZWNvcmRcIiwgKytjb3VudCk7XG4gIH1cbjwvc2NyaXB0PiIsIk5lc3RlZC52dWUiOiI8dGVtcGxhdGU+XG4gIDxjb21wb25lbnQgOmlzPVwiY29tcFwiIHYtYmluZD1cInsgLi4ucHJvcHMsIC4uLiRhdHRycyB9XCIgLz5cbjwvdGVtcGxhdGU+XG5cbjxzY3JpcHQ+XG5leHBvcnQgZGVmYXVsdCB7XG4gIGluaGVyaXRBdHRycyA6IGZhbHNlXG59XG48L3NjcmlwdD5cblxuPHNjcmlwdCBzZXR1cD5cbiAgaW1wb3J0IElubmVyIGZyb20gJy4vSW5uZXIudnVlJztcbiAgXG4gIGNvbnN0IGNvbXAgPSBJbm5lcjtcbiAgY29uc3QgcHJvcHMgPSBkZWZpbmVQcm9wcygpO1xuICBcbiAgZGVmaW5lRW1pdHMoW1widXBkYXRlOnJlY29yZFwiXSk7XG4gIFxuPC9zY3JpcHQ+In0=) example for the 'broken' state.  Clicking the button should log a message from the event handle in App.vue but it doesn't.  Comment out the `defineEmits` line in Nested.vue and it does.

Everything works at runtime, I just need a way to tell `vue-tsc` that I know `@row-double-clicked` is safe in this case?

**Comment by johnsoncodehk** [comment]
Thanks for your reproduce case, but `@row-double-clicked` should be unknown if don't have deineEmit, what error you got? If you have special tsconfig settings please show it.

<img width="255" alt="螢幕截圖 2021-09-03 上午11 43 32" src="https://user-images.githubusercontent.com/16279759/131947453-f2f75c7b-e308-483d-ba85-b24122d491ac.png">

**Reply by RobWalker** [reply]
Right -- it is unknown, I was hoping for an 'escape' hatch in the syntax that would let me use the equivalent of an `any` or something to say I was aware of that in this case and to not worry about it.

I have some components that can wrap any other component within them so can't know statically what the valid emits are.

It looks like doing:
```
<ServerReport  v-on="events">
...
const events: any = { 'row-double-clicked': onSelect };
```

is working to get past this though with just one extra line, so that works for me!

Thanks.
